From f1183b6c69df7c058a737ae7d3a5a0185e41d076 Mon Sep 17 00:00:00 2001
From: Gwenhael Goavec-Merou <gwenhael.goavec-merou@trabucayre.com>
Date: Thu, 15 Jun 2023 09:40:24 -0400
Subject: [PATCH] ohwr-build-script: - fix wrtd-ref-designs repo spec adc
 ressources - fix scripts: install wrtd-ref-designs software and reorder
 scripts

---
 patches/wrtd-ref-designs.diff            | 42 ++++++++++++++++++++++++++++++++
 scripts/wrtd_ref_designs_install.sh      |  5 ++++
 scripts/wrtd_ref_spec150t_adc_install.sh |  2 +-
 3 files changed, 48 insertions(+), 1 deletion(-)
 create mode 100644 patches/wrtd-ref-designs.diff

diff --git a/patches/wrtd-ref-designs.diff b/patches/wrtd-ref-designs.diff
new file mode 100644
index 0000000..1bdb342
--- /dev/null
+++ b/patches/wrtd-ref-designs.diff
@@ -0,0 +1,42 @@
+diff --git a/software/drivers/wrtd-ref-spec150t-adc-core.c b/software/drivers/wrtd-ref-spec150t-adc-core.c
+index 61b2081..5f2c9e7 100644
+--- a/software/drivers/wrtd-ref-spec150t-adc-core.c
++++ b/software/drivers/wrtd-ref-spec150t-adc-core.c
+@@ -12,6 +12,8 @@
+ #include <linux/mod_devicetable.h>
+ 
+ enum wrtd_spec150t_adc_dev_offsets {
++	FA_SPEC_DBL_ADC_META_START = 0x00000000,
++	FA_SPEC_DBL_ADC_META_END   = 0x00000040,
+ 	WRTD_SPEC150T_ADC_FA100_MEM_START = 0x00002000,
+ 	WRTD_SPEC150T_ADC_FA100_MEM_END = 0x00003A00,
+ 	WRTD_SPEC150T_ADC_TRTL_MEM_START = 0x0001E000,
+@@ -36,6 +38,18 @@ static struct resource wrtd_spec150t_adc_fa100_res[] = {
+ 		.start = 0,
+ 		.end = 0,
+ 	},
++	 {
++		.name = "fmc-adc-dma",
++		.flags = IORESOURCE_DMA,
++	},
++
++	{
++	.name = "fmc-adc-meta",
++	.flags = IORESOURCE_MEM,
++	.start = FA_SPEC_DBL_ADC_META_START,
++	.end = FA_SPEC_DBL_ADC_META_END,
++	},
++
+ };
+ 
+ static struct resource wrtd_spec150t_adc_trtl_res[] = {
+@@ -72,7 +86,8 @@ static const struct mfd_cell wrtd_spec150t_adc_mfd_devs[] = {
+ 		.resources = wrtd_spec150t_adc_trtl_res,
+ 	},
+ 	[WRTD_SPEC150T_ADC_MFD_FA100] = {
+-		.name = "adc-100m-spec",
++		//.name = "adc-100m-spec",
++		.name = "fmc-adc-100m",
+ 		.platform_data = NULL,
+ 		.pdata_size = 0,
+ 		.num_resources = ARRAY_SIZE(wrtd_spec150t_adc_fa100_res),
diff --git a/scripts/wrtd_ref_designs_install.sh b/scripts/wrtd_ref_designs_install.sh
index 426c8c6..3d2f5af 100644
--- a/scripts/wrtd_ref_designs_install.sh
+++ b/scripts/wrtd_ref_designs_install.sh
@@ -11,6 +11,11 @@ cd $BUILD_DIR
 [ -d $PROJECT ] || git clone https://ohwr.org/project/wrtd-ref-designs.git $PROJECT
 cd $PROJECT
 git checkout -B release v2.0.0
+[ $SKIP_PATH ] || [ -e $PATCH_DIR/$PROJECT.diff ] && patch -p1 < $PATCH_DIR/$PROJECT.diff
+export FMC=$BUILD_DIR/fmc-sw
+export ADC_ABS=$BUILD_DIR/fmc-adc-100m14b4cha
+[ $SKIP_BUILD ] || make -C software/drivers
+[ $SKIP_INSTALL ] || sudo -E make -C software/drivers modules_install
 [ $SKIP_INSTALL ] || sudo -E install -d /etc/udev/rules.d/
 [ $SKIP_INSTALL ] || sudo -E install -m 0644 software/udev/rules.d/* /etc/udev/rules.d/
 [ $SKIP_INSTALL ] || sudo -E sh -c 'getent group wrtd 2>&1 > /dev/null || groupadd wrtd'
diff --git a/scripts/wrtd_ref_spec150t_adc_install.sh b/scripts/wrtd_ref_spec150t_adc_install.sh
index a11014f..51ef93a 100644
--- a/scripts/wrtd_ref_spec150t_adc_install.sh
+++ b/scripts/wrtd_ref_spec150t_adc_install.sh
@@ -4,8 +4,8 @@ PROJECT=wrtd_ref_spec150t_adc
 
 . ./pre_script
 
-bash $SCRIPT_DIR/wrtd_ref_designs_install.sh
 bash $SCRIPT_DIR/spec_carrier_build.sh
 bash $SCRIPT_DIR/adc_lib_build.sh
+bash $SCRIPT_DIR/wrtd_ref_designs_install.sh
 
 . $SCRIPT_DIR/post_script
-- 
1.8.3.1

